src/main/java/org/swisspush/redisques/QueueStatsService.java

```diff
@@ -97,10 +98,13 @@ public <CTX> void getQueueStats(CTX mCtx, GetQueueStatsMentor<CTX> mentor) {
                 for (Queue q : req1.queues) req1.queueNames.add(q.name);
                 fetchRetryDetails(req1, (ex2, req2) -> {
                     if (ex2 != null) { onDone.accept(ex2, null); return; }
-                    attachDequeueStats(req2, (ex3, req3) -> {
-                        if (ex3 != null) { onDone.accept(ex3, null); return; }
-                        onDone.accept(null, req3.queues);
-                    });
+                    if (fetchQueueStats) {
```

Comment:
There is a problem here with `onDone.accept(...)` called too many times. You can verify this in the code of `onDone` where an Exception is thrown (or logged) when called more than once.

Reply:
I will implement in another function as you suggest in our call. Thanks.